### DISCUSSION TITLE
Nushell `quitcd` script doesn't work

### DISCUSSION THREAD

**Discussion by arjpar** [jarun/nnn#1605] [Q&A] [answered]
When pasting the code found here [misc/quitcd/quitcd.nu](https://github.com/jarun/nnn/blob/master/misc/quitcd/quitcd.nu) into either `config.nu` or `env.nu`, it does not give the desired effect of CD-ing once quit.

Is anyone else having this issue or is it just me?

**Comment by N-R-K** [comment]
I don't use nutshell and I'm not sure if anyone else here does or not. Perhaps the author of that script (@mmai) can help you out.

**Comment by mmai** [comment]
Hi @arjpar. Do you launch nnn with the `n` shortcut  ?
If so, could you add the following two `# debug`  lines, start a new nushell, and show the output after quiting a `n` session ?  

```
  ^nnn -eda ($x | str join)

  echo $env.NNN_TMPFILE  # debug
  open $env.NNN_TMPFILE   # debug
  let newpath = (
```

**Reply by arjpar** [reply]
I added those two lines, and it doesn't output anything. It seems like it isn't even being called.

**Reply by arjpar** [reply]
> I added those two lines, and it doesn't output anything. It seems like it isn't even being called.

Ooh, so I found out why it doesn't run. I previously had an alias mapping the `n` command to `nnn`. After removing that, now it does run, but it said that the `$"($cfgHome)/.config/nnn/.lastd"` directory (defined in `let-env NNN_TMPFILE`) doesn't exist. Upon `echo`ing `$"($cfgHome)/.config/nnn/.lastd"` it gave me `/Users/z/.config/.config/nnn/.lastd`, which is the incorrect directory path. So something is wrong in either the definition of `cfgHome` or `NNN_TMPFILE`. Once you fix that, the problem is solved.

**Reply by mmai** [reply]
Thank you @arjpar . The problem is indeed in the definition of both cfgHome and NNN_TMPFILE when the XDG_CONFIG_HOME variable is set ( this was not the case on my system so I didn't see it). I will make a PR with a fix.

**Comment by arjpar** [comment, marked as answer]
Problem solved. See [my comment](https://github.com/jarun/nnn/discussions/1605#discussioncomment-5221559) for an answer.
